tests/interfaces/python/test_5_tcopflow.py

```diff
@@ -0,0 +1,40 @@
+import exago
+from mpi4py import MPI
+import tempfile
+import os
+import shutil
+import pytest
+from check_preconditions import check_preconditions
+import mpi4py.rc
+mpi4py.rc.threads = False
+
+check_preconditions()
```

Comment:
```suggestion
from mpi4py import MPI
import tempfile
import os
import shutil
import pytest
from check_preconditions import check_preconditions
import mpi4py.rc
mpi4py.rc.threads = False

check_preconditions()
import exago
```

@Jayapreethi can you try out this change?

You had an error where you were running something along the lines of:

- `cmake -B build -S .`
- `cmake --build build -- -j 8`
- `cd build && ctest -VV -R Python`

And were getting errors with `import exago`. The whole point of the `check_preconditions()` functions seems to be to actually print some helpful debug messages, so I think moving that to before `import exago` would make the error message much more friendly.

Please also extend this change into other tests if you verify that it helps and works as I have described.

Reply:
We navigated this offline... Still need to apply to all files